Kconfig

```diff
@@ -50,6 +50,17 @@ config IMAGE_BOOT_PG2
 	  that it fits AND you have an external flash programmer at hand that
 	  allows to recover. Otherwise you risk to BRICK THE IOT2050!
 
+config IMAGE_BOOT_M2
+	bool "Firmware image for M2 devices"
+	help
+	  Build the firmware image that is responsible for booting Product
+	  Generation 2 (PG2) devices.
+	  It's the same as IMAGE_BOOT_PG2 option
+
+	  WARNING: Do not flash this image onto your device unless you know
+	  that it fits AND you have an external flash programmer at hand that
+	  allows to recover. Otherwise you risk to BRICK THE IOT2050!
+
```

Comment:
We need reconsider the whole variant identification system. PG1/PG2/M.2 are very our point of view, we need something from customer's, such as the FS_TYPE or something that easy found on the device to distinguish them.

Reply:
At least, we need to reference THAT information somewhere as well, even if we keep the codename.